src/macaron/repo_finder/repo_finder_deps_dev.py

```diff
@@ -70,51 +72,53 @@ def find_repo(self, purl: PackageURL) -> str:
 
         return ""
 
-    def _create_urls(self, namespace: str, name: str, version: str, type_: str) -> list[str]:
+    def _create_urls(self, purl: PackageURL) -> list[str]:
         """
         Create the urls to search for the metadata relating to the passed artifact.
 
         If a version is not specified, remote API calls will be used to try and find one.
 
         Parameters
         ----------
-        namespace : str
-            The PURL namespace.
-        name: str
-            The PURL name.
-        version: str
-            The PURL version.
-        type : str
-            The PURL type.
+        purl : PackageURL
+            The PURL of an artifact.
 
         Returns
         -------
         list[str]
             The list of created URLs.
         """
-        base_url = self._create_type_specific_url(namespace, name, type_)
+        # See https://docs.deps.dev/api/v3alpha/
+        base_url = f"https://api.deps.dev/v3alpha/purl/{encode(str(purl)).replace('/', '%2F')}"
 
         if not base_url:
             return []
 
-        if version:
-            return [f"{base_url}/versions/{version}"]
+        if purl.version:
+            return [base_url]
 
         # Find the latest version.
         response = send_get_http_raw(base_url, {})
 
         if not response:
             return []
 
-        metadata = json.loads(response.text)
-        versions = metadata["versions"]
-        latest_version = versions[len(version) - 1]["versionKey"]["version"]
+        try:
+            metadata: dict = json.loads(response.text)
+        except ValueError as error:
+            logger.debug("Failed to parse response from deps.dev: %s", error)
+            return []
 
-        if latest_version:
-            logger.debug("Found latest version: %s", latest_version)
-            return [f"{base_url}/versions/{latest_version}"]
+        try:
+            versions_keys = ["package", "versions"] if "package" in metadata else ["version"]
```

Comment:
nit: I think this line can be move outside of the try catch block because it doesn't raise `JsonError`.